microsoft/testsuites/hpc/infiniand.py

```diff
@@ -0,0 +1,125 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from assertpy import assert_that
+
+from lisa import Logger, Node, TestCaseMetadata, TestSuite, TestSuiteMetadata
+from lisa.features import NetworkInterface
+from lisa.tools import Modprobe
+from lisa.util import SkippedException
+
+
+@TestSuiteMetadata(
+    area="core",
+    category="functional",
+    description="""
+    Tests the functionality of infiniband.
+    """,
+)
+class Infiniband(TestSuite):
+    @TestCaseMetadata(
+        description="""
+        This test case will
+        1. Determine whether the VM is IB over SR-IOV
+        2. Ensure waagent is configures with OS.EnableRDMA=y
+        3. Check that appropriate drivers are present
+        """,
+        priority=2,
+    )
+    def verify_hpc_over_sriov(self, log: Logger, node: Node) -> None:
+
+        if node.execute("test -d /sys/class/infiniband", shell=True).exit_code != 0:
+            raise SkippedException(
+                "/sys/class/infiniband/ not found. "
+                "Only HPC machines are supported by this test."
+            )
+
+        nic = node.features[NetworkInterface]
+        if (
+            node.execute(
+                'lspci | grep "Virtual Function"', shell=True, sudo=True
+            ).exit_code
+            != 0
+            or not nic.is_enabled_sriov()
+        ):
+            raise SkippedException("Inifiniband over SR-IOV was not detected.")
+
+        result = node.execute(
```

Comment:
Add this logic in waagent as `is_rdma_enabled`, and call its `get_configuration`. The file is `/lisa/sut_orchestrator/tools.py`.